File: solutions/013.Roman_to_Integer/AC_simulation_1.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_simulation_1.py
# Create Date: 2015-03-02 23:19:56
# Usage:       AC_simulation_1.py 
# Descripton:  


class Solution:
    # @return an integer
    def romanToInt(self, s):
        val = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        ret = 0
        for i in range(len(s)):
            if i > 0 and val[s[i]] > val[s[i - 1]]:
                ret += val[s[i]] - 2 * val[s[i - 1]]
            else:
                ret += val[s[i]]
        return ret

File: solutions/225.Implement_Stack_using_Queues/AC_simulation_1.java
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_simulation_1.java
*  Create Date: 2015-07-25 12:22:04
*  Descripton:   
*/

import java.util.LinkedList;
import java.util.Queue;

class MyStack {
    private Queue queue;

    // Push element x onto stack.
    public void push(int x) {
        Queue q = new LinkedList();
        q.add(x);
        q.add(queue);
        queue = q;
    }

    // Removes the element on top of the stack.
    public void pop() {
        queue.remove();
        queue = (Queue)queue.peek();
    }

    // Get the top element.
    public int top() {
        return (int)queue.peek();
    }

    // Return whether the stack is empty.
    public boolean empty() {
        return queue == null;
    }
}